gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/Job.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.bigquery;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A Google BigQuery Job.
+ *
+ * <p>Objects of this class are immutable. To get a {@code Job} object with the most recent
+ * information use {@link #reload}.
+ * </p>
+ */
+public final class Job {
+
+  private final BigQuery bigquery;
+  private final JobInfo info;
+
+  /**
+   * Constructs a {@code Job} object for the provided {@code JobInfo}. The BigQuery service
+   * is used to issue requests.
+   *
+   * @param bigquery the BigQuery service used for issuing requests
+   * @param info jobs's info
+   */
+  public Job(BigQuery bigquery, JobInfo info) {
+    this.bigquery = checkNotNull(bigquery);
+    this.info = checkNotNull(info);
+  }
+
+  /**
+   * Creates a {@code Job} object for the provided job's user-defined id. Performs an RPC call
+   * to get the latest job information.
+   *
+   * @param bigquery the BigQuery service used for issuing requests
+   * @param job job's id, either user-defined or picked by the BigQuery service
+   * @return the {@code Job} object or {@code null} if not found.
+   * @throws BigQueryException upon failure
+   */
+  public static Job load(BigQuery bigquery, String job) {
+    JobInfo info = bigquery.getJob(job);
+    return info != null ? new Job(bigquery, info) : null;
+  }
+
+  /**
+   * Returns the job's information.
+   */
+  public JobInfo info() {
+    return info;
+  }
+
+  /**
+   * Checks if this job exists.
+   *
+   * @return {@code true} if this job exists, {@code false} otherwise.
+   * @throws BigQueryException upon failure
+   */
+  public boolean exists() {
+    return bigquery.getJob(info.jobId(), BigQuery.JobOption.fields()) != null;
+  }
+
+  /**
+   * Checks if this job has completed its execution, either failing or succeeding.
+   *
+   * @return {@code true} if this job is in {@link JobStatus.State#DONE} state, {@code false} if the
```

Comment:
A job can not be deleted. It can be cancelled but the job's resource will still be returned with `state = DONE`. I agree that we can suggest to use `while(job.exists() && !job.isDone())`.